tests/test_harvesters.py

```diff
@@ -191,33 +191,35 @@ def test_creodias_harvester(self):
 
-    def test_osisaf_harvester_extra_excludes(self):
-        """ extra excludes should have passed by the excludes as a list in configuration file.
+    def test_osisaf_harvester_include(self):
+        """ include criteria should have passed by the "includes" as a list in configuration file.
         Otherwise, accossiated error must be raised """
         harvester = harvesters.OSISAFHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1,
-                                               excludes=['ease', '_sh_polstere', ])
-        self.assertListEqual(harvester._current_crawler.excludes,
-                             crawlers.ThreddsCrawler.EXCLUDE + ['ease', '_sh_polstere'])
+                                               includes='ease|_sh_polstere')
+        self.assertEqual(harvester._current_crawler.include, re.compile('ease|_sh_polstere'))
         harvester = harvesters.OSISAFHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1)
-        self.assertListEqual(harvester._current_crawler.excludes, crawlers.ThreddsCrawler.EXCLUDE)
-
+        self.assertIsNone(harvester._current_crawler.include)
         with self.assertRaises(HarvesterConfigurationError):
             harvester = harvesters.OSISAFHarvester(
-                urls=[''], max_fetcher_threads=1, max_db_threads=1, excludes='ease')
+                urls=[''], max_fetcher_threads=1, max_db_threads=1, includes=['ease'])
 
-    def test_extra_excludes_with_no_CLASS_EXCLUDE(self):
-        """ shall return the excludes from the config file """
+    def test_excludes_with_and_without_CLASS_EXCLUDE(self):
+        """ shall return the proper excludes in harvester from the class EXCLUDEs of crawler """
         class TestCrawler(crawlers.WebDirectoryCrawler):
             EXCLUDE = None
+        class TestHarvester(harvesters.WebDirectoryHarvester):
+            ingester = ingesters.DDXIngester
+            crawler = TestCrawler
+        harvester = TestHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1)
+        self.assertIsNone(harvester._current_crawler.EXCLUDE)
 
+        class TestCrawler(crawlers.WebDirectoryCrawler):
+            EXCLUDE = re.compile('excluding_criteria')
         class TestHarvester(harvesters.WebDirectoryHarvester):
             ingester = ingesters.DDXIngester
             crawler = TestCrawler
-        harvester = TestHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1,
-                                  excludes=['ease', '_sh_polstere', ])
-        self.assertListEqual(harvester._current_crawler.excludes, ['ease', '_sh_polstere'])
-        harvester = TestHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1,)
-        self.assertEqual(harvester._current_crawler.excludes, [])
+        harvester = TestHarvester(urls=[''], max_fetcher_threads=1, max_db_threads=1)
+        self.assertEqual(harvester._current_crawler.EXCLUDE, re.compile('excluding_criteria'))
```

Comment:
What do you test here? The mechanism to merge exclude lists does not exist anymore, so it looks to me like you can remove this test.

Reply:
I am testing this attribute to set in the harvester. It must be a regex or None. anything except these two should be erroneous. These are harvesters tests and mainly dealing with correct way of object creation from those classes. Actually, these tests help developers not to destroy sth in the middle of their development.

Reply:
> I am testing this attribute to set in the harvester.
The harvester just instantiates a crawler, it does nothing to the class attributes of the crawler.

> It must be a regex or None.
That's decided in the Crawler class definition, so I don't see what you can test here.

Reply:
I am testing that the `EXCLUDE` attribute of created objects of harvester have what the developer define in the crawler class definition. It matters to have that `EXCLUDE` attribute . Because it must be correctly instantiated with it.

Reply:
It's a class attribute, all WebDirectoryCrawlers will have it.

If you want another argument, why do you test this particular class attribute and not all the others?

Reply:
Yes, that is the point! I want that attribute to be in created obj for ANY class. That is the reason why I define a new test class for this test. We must see that EXCLUDE in the `harvester._current_crawler.EXCLUDE` of any harvester. That is testing the instantiation which is done in `harvesters.py` and tested in `test_harvesters.py`

Reply:
Discussed internally and the test is deleted at the end.